### PR TITLE
QMPMonitor: support bitmap persistent

### PR DESCRIPTION
**Pull request by lolyu**
id:1718821
defined three methods to add, remove, and clear bitmap:
add_block_dirty_bitmap
remove_block_dirty_bitmap
clear_block_dirty_bitmap
Those three functions delegate call to _operate_dirty_bitmap.

Signed-off-by: lolyu <lolyu@redhat.com>

### CODE REVIEW

**Comment by lolyu** [comment]
@luckyh updated, please review.

**Comment by lolyu** [comment]
```
 (1/6) Host_RHEL.m7.u7.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.0.x86_64.io-github-autotest-qemu.live_backup_add_bitmap.with_qcow2.persistent_on: PASS (76.60 s)
 (2/6) Host_RHEL.m7.u7.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.0.x86_64.io-github-autotest-qemu.live_backup_add_bitmap.with_qcow2.persistent_off: PASS (75.52 s)
 (3/6) Host_RHEL.m7.u7.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.0.x86_64.io-github-autotest-qemu.live_backup_add_bitmap.with_qcow2.persistent_default: PASS (77.11 s)
 (4/6) Host_RHEL.m7.u7.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.0.x86_64.io-github-autotest-qemu.live_backup_add_bitmap.with_raw.persistent_on: PASS (49.43 s)
 (5/6) Host_RHEL.m7.u7.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.0.x86_64.io-github-autotest-qemu.live_backup_add_bitmap.with_raw.persistent_off: PASS (75.63 s)
 (6/6) Host_RHEL.m7.u7.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.0.x86_64.io-github-autotest-qemu.live_backup_add_bitmap.with_raw.persistent_default: PASS (77.20 s)
```

File: virttest/qemu_monitor.py [hunk 1]
```diff
@@ -2713,18 +2713,58 @@ def operate_dirty_bitmap(self, operation, node, name, granularity=65536):
...
+        Add a dirty bitmap.
+
+        :param node: device node
+        :param name: bitmap name
+        :param granularity: segment size
+        :param persistent: persistent through QEMU shutdown
+        """
+        return self._operate_dirty_bitmap("add",
+                                          **{k: v for k, v in locals().items()
+                                             if k != "self" and v is not None})
```
Comment (luckyh) [inline virttest/qemu_monitor.py]: Request to pass the arguments directly instead of doing the iteration like this.

**Comment by lolyu** [comment]
@luckyh updated.

File: virttest/qemu_monitor.py [hunk 2]
```diff
@@ -1725,6 +1725,13 @@ def cmd(self, cmd, args=None, timeout=CMD_TIMEOUT, debug=True, fd=None):
         finally:
             self._lock.release()
 
+    @staticmethod
+    def _build_args(**kargs):
+        """
+        Build args used in cmd.
+        """
+        return {k: v for k, v in kargs.items() if v is not None}
```
Comment (luckyh) [inline virttest/qemu_monitor.py]: There is another thing should be done within this function: to convert the arguments' name by replacing `_` to `-`, where the former is accepted by python while the latter is accepted by QMP.

**Comment by hereischen** [comment]
Acked-by: Haotong Chen hachen@redhat.com

**Comment by luckyh** [comment]
Thanks all.

**Pull request merged**